src/GetText.Extractor/Program.cs

```diff
@@ -25,27 +26,46 @@ private static async Task<int> Main(string[] args)
+            Option aliasPlural = CommandLineOptions.GetPluralStringAliases;
+            Option aliasDefinitePlural = CommandLineOptions.GetParticularPluralStringAliases;
+
+            aliasString.AllowMultipleArgumentsPerToken = true;
+            aliasDefinite.AllowMultipleArgumentsPerToken = true;
+            aliasPlural.AllowMultipleArgumentsPerToken = true;
+            aliasDefinitePlural.AllowMultipleArgumentsPerToken = true;
 
             rootCommand.Add(sourceOption);
             rootCommand.Add(outFile);
             rootCommand.Add(verbose);
             rootCommand.Add(unixPathSeparator);
             rootCommand.Add(sort);
+            rootCommand.Add(aliasString);
+            rootCommand.Add(aliasDefinite);
+            rootCommand.Add(aliasPlural);
+            rootCommand.Add(aliasDefinitePlural);
 
-            rootCommand.SetHandler(async (IList<FileInfo> sources, FileInfo target, bool unixstyle, bool sortOutput, bool verbose) =>
+            rootCommand.SetHandler(async (IList<FileInfo> sources, FileInfo target, bool unixstyle, bool sortOutput, bool verbose,
+                List<string> @as, List<string> ad, List<string> ap, List<string> adp) =>
             {
-                await Execute(sources, target, unixstyle, sortOutput, verbose).ConfigureAwait(false);
-            }, sourceOption, outFile, unixPathSeparator, sort, verbose);
+                await Execute(sources, target, unixstyle, sortOutput, verbose, @as, ad, ap, adp).ConfigureAwait(false);
+            }, sourceOption, outFile, unixPathSeparator, sort, verbose, aliasString, aliasDefinite, aliasPlural, aliasDefinitePlural);
 
             return await rootCommand.InvokeAsync(args).ConfigureAwait(false);
 
         }
 
-        private static async Task Execute(IList<FileInfo> sources, FileInfo target, bool unixStyle, bool sortOutput, bool verbose)
+        private static async Task Execute(IList<FileInfo> sources, FileInfo target, bool unixStyle, bool sortOutput, bool verbose,
+            List<string> getStringAliases, List<string> getParticularStringAliases, List<string> getPluralStringAliases, List<string> getParticularPluralStringAliases)
         {
             catalog = new CatalogTemplate(target.FullName);
 
-            SyntaxTreeParser parser = new SyntaxTreeParser(catalog, sources, unixStyle, verbose);
+            SyntaxTreeParser parser = new SyntaxTreeParser(catalog, sources, unixStyle, verbose, new Aliases()
+            {   GetString = new List<string>() { "GetString" }.Concat(getStringAliases).ToList(),
+                GetParticularString = new List<string>() { "GetParticularString" }.Concat(getParticularStringAliases).ToList(),
+                GetPluralString = new List<string>() { "GetPluralString" }.Concat(getPluralStringAliases).ToList(),
+                GetParticularPluralString = new List<string>() { "GetParticularPluralString" }.Concat(getParticularPluralStringAliases).ToList() });
```

Comment:
please keep the default aliases out of program and where they were (ParserBase), such as
```suggestion
 in Engine\ParserBase.cs
        public ParserBase(CatalogTemplate catalog, bool unixStyle, bool verbose, Aliases aliases)
        {
.....
            GetStringAliases = new List<string>() {"GetString"}.AddRange(aliases.GetString ?? Enumerable.Empty<string>()).ToHashSet(); // etc for other aliases

Reply:
The only difference was that AddRange doesn't work in that syntax, as it modifies the list and returns nothing. Other than that, I basically copied your suggestions exactly.